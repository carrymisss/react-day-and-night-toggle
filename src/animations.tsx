import { keyframes } from 'styled-components';

export const growbrighter = keyframes`
    0% {
        box-shadow: 0 0 4px 0px rgb(255, 255, 255);
    }
    100% {
        box-shadow: 0 0 15px 0px rgb(255, 255, 255);
    }
`;

export const startravel1 = keyframes`
    0% {
        transform: rotate(65deg) translate(-80px) scale(1.0);
    }
    100% {
        transform: rotate(-60deg) translate(-80px) scale(1.3);
    }
`;

export const startravel2 = keyframes`
    0% {
        transform: rotate(27deg) translate(-160px) scale(1.6);
    }
    100% {
        transform: rotate(-25deg) translate(-160px) scale(1.2);
    }
`;

export const startravel3 = keyframes`
    0% {
        transform: rotate(22deg) translate(-200px) scale(1.2);
    }
    100% {
        transform: rotate(-20deg) translate(-200px) scale(0.7);
    }
`;
export const shootingstartravel = keyframes`
    0% {
        transform: translate(200px, -30px) rotate(-30deg);
    }
    8.33% {
        transform: translate(-180px, 170px) rotate(-30deg);
    }
    100% {
        transform: translate(-180px, 170px) rotate(-30deg);
    }
`;

export const growtrail = keyframes`
    0% {
        width: 40px;
    }
    8.33% {
        width: 180px;
    }
    100% {
        width: 180px;
    }
`;

export const cloudtravel1 = keyframes`
    0% {
        left: -80px;
        transform: scale(1);
    }
    100% {
        left: 330px;
        transform: scale(0.8);
    }
`;

export const cloudtravel2 = keyframes`
    0% {
        left: -90px;
        transform: scale(0.6);
    }
    100% {
        left: 330px;
        transform: scale(1);
    }
`;

export const cloudtravel3 = keyframes`
    0% {
        left: -100px;
        transform: scale(0.7);
    }
    100% {
        left: 340px;
        transform: scale(0.65);
    }
`;
