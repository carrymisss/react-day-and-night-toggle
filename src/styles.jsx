import styled, { css } from 'styled-components';
import { cloudtravel1, cloudtravel2, cloudtravel3, growbrighter, growtrail, shootingstartravel, startravel1, startravel2, startravel3, } from './animations';
export const SwitchContainer = styled.div `
  position: relative;
  height: ${(props) => (props.size % 2 === 1 ? props.size + 0.1 : props.size)}px;
  width: ${(props) => ((props.size * 2.5) % 2 === 1 ? props.size * 2.5 + 1 : props.size * 2.5)}px;
`;
export const Switch = styled.div `
  box-sizing: border-box;
  cursor: pointer;
  position: absolute;
  width: 300px;
  height: 120px;
  left: 50%;
  top: 50%;
  border-radius: 70px;
  border: 5px solid ${(props) => (props.checked ? 'rgb(40, 40, 51)' : 'rgb(211, 239, 255)')};
  background-color: ${(props) => (props.checked ? 'rgb(23, 23, 35)' : 'rgb(186, 230, 255)')};
  transition:
    border-color 1500ms,
    background-color 1500ms,
    box-shadow 1500ms;
  ${(props) => props.shadows && `box-shadow: 0 0 35px 1px ${props.checked ? '#E0E0E0' : '#616161'};`}
  transform: translate(-50%, -50%) scale(${(props) => props.size * 0.00832});
  ${(props) => props.checked &&
    css `
      ${Circle} {
        width: 100px;
        height: 100px;
        left: 66%;
        background-color: rgb(237, 242, 241);
        box-shadow: 0 0 20px 3px rgb(244, 247, 246);
        &::before,
        &::after {
          opacity: 1;
          transition: opacity 2s 0.2s;
        }
      }
      ${Star} {
        opacity: 1;
        transition: opacity 2s 1.5s;
      }
      ${Stars} {
        & ${Star} {
          &:nth-of-type(1) {
            animation: ${css `
              ${startravel1} 15s linear infinite
            `};
            animation-delay: 0.4s;
          }
          &:nth-of-type(2) {
            animation: ${css `
              ${startravel2} 12s linear infinite
            `};
            animation-delay: 0.2s;
          }
          &:nth-of-type(3) {
            animation: ${css `
              ${startravel3} 18s linear infinite
            `};
            animation-delay: 0s;
          }
        }
      }
      ${ShootingStarWrapper} {
        opacity: 1;
        transition: opacity 0s 2s;
        animation: ${css `
          ${shootingstartravel} 24s linear infinite
        `};
        animation-delay: 6s;
      }
      ${ShootingStar} {
        &::after {
          animation: ${css `
            ${growtrail} 24s linear infinite
          `};
          animation-delay: 6s;
        }
      }
      ${Cloud} {
        opacity: 0;
        transition: opacity 1s 0s;
        &:nth-of-type(1),
        &:nth-of-type(2),
        &:nth-of-type(3) {
          animation: none;
        }
      }
    `}
`;
export const ContentWrapper = styled.div `
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  border-radius: 60px;
  -webkit-border-radius: 60px;
  overflow: hidden;
  z-index: 2;
`;
export const Circle = styled.div `
  position: absolute;
  width: 70px;
  height: 70px;
  left: 0%;
  top: 50%;
  transform: translate(0%, -50%);
  border-radius: 50%;
  background-color: rgb(255, 233, 183);
  box-shadow:
    0 0 20px 10px rgb(255, 233, 183),
    0 0 40px 30px rgb(255, 219, 140),
    0 0 90px 50px rgb(255, 209, 109);
  transition:
    left 2s,
    transform 2s,
    width 2s,
    height 2s,
    box-shadow 2s,
    background-color 2s 0.2s;
  z-index: 1;

  &::before,
  &::after {
    content: '';
    position: absolute;
    border-radius: 50%;
    opacity: 0;
    transition: opacity 0s;
  }

  &::before {
    left: 13%;
    top: 30%;
    width: 12px;
    height: 20px;
    background-color: rgb(183, 188, 187);
    box-shadow:
      0 0 4px 1px rgb(183, 188, 187),
      0 0 10px 6px rgb(192, 198, 197);
  }

  &::after {
    left: 60%;
    top: 75%;
    width: 12px;
    height: 12px;
    background-color: rgb(183, 188, 187);
    box-shadow:
      0 0 3px 1px rgb(183, 188, 187),
      0 0 5px 4px rgb(192, 198, 197);
  }
`;
export const Stars = styled.div ``;
export const Star = styled.div `
  position: absolute;
  width: 6px;
  height: 6px;
  left: 80%;
  top: 50%;
  border-radius: 50%;
  background-color: white;
  box-shadow: 0 0 5px 0px rgb(255, 255, 255);
  animation: none;
  opacity: 0;

  &::before,
  &::after {
    content: '';
    position: absolute;
    transform: translate(-50%, -50%);
    border-radius: 50%;
    background-color: white;
    box-shadow: 0 0 4px 0px rgb(255, 255, 255);
    animation: ${css `
      ${growbrighter} 1s linear alternate infinite
    `};
    left: 50%;
    top: 50%;
  }

  &::before {
    width: 20px;
    height: 1px;
  }

  &::after {
    width: 1px;
    height: 20px;
  }
`;
export const ShootingStarWrapper = styled.div `
  position: absolute;
  transform: translate(200px, -34px) rotate(-30deg);
  opacity: 0;
`;
export const ShootingStar = styled.div `
  position: absolute;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  left: 20%;
  top: 50%;
  background: radial-gradient(rgba(255, 231, 193, 0.9) 30%, rgba(237, 155, 61, 1));
  box-shadow:
    -5px 0 20px 2px rgba(237, 155, 61, 0.7),
    0 0 15px 5px rgba(237, 155, 61, 0.5);

  &::after {
    content: '';
    position: absolute;
    width: 160px;
    height: 15px;
    border-top-right-radius: 70%;
    border-bottom-right-radius: 70%;
    border-top-left-radius: 10px;
    border-bottom-left-radius: 10px;
    top: 50%;
    transform: translate(0, -50%);
    background: radial-gradient(rgba(255, 231, 193, 0.9) 1%, rgba(237, 155, 61, 0.9));
    box-shadow:
      10px 0 10px 1px rgba(237, 155, 61, 0.5),
      0px 3px 15px 5px rgba(237, 155, 61, 0.5),
      0px -3px 15px 5px rgba(237, 155, 61, 0.5);
    z-index: -1;
  }
`;
export const Cloudpart = styled.div ``;
export const Clouds = styled.div ``;
export const Cloud = styled.div `
  &:nth-of-type(1),
  &:nth-of-type(2),
  &:nth-of-type(3) {
    position: absolute;
    border-radius: 50%;
    background: radial-gradient(rgb(255, 255, 255) 30%, rgb(230, 230, 230));
    z-index: 2;
    opacity: 1;
    transition: opacity 2s 0.4s;
    &::before,
    &::after {
      content: '';
      position: absolute;
      border-radius: 50%;
      background: inherit;
      z-index: 3;
    }
    & ${Cloudpart}:nth-of-type(1),
    & ${Cloudpart}:nth-of-type(2) {
      content: '';
      position: absolute;
      border-radius: 50%;
      background: inherit;
      z-index: 2;
      bottom: 0%;
      &::after,
      &::before {
        content: '';
        position: absolute;
        border-radius: 50%;
        background: inherit;
        z-index: 3;
        top: 50%;
      }
    }
  }
  &:nth-of-type(1) {
    width: 50px;
    height: 55px;
    left: -80px;
    top: 5px;
    animation: ${css `
      ${cloudtravel1} 22s linear infinite
    `};
    animation-delay: 5.4s;
    &::before {
      width: 30px;
      height: 20px;
      left: -55%;
      top: 70%;
      transform: translate(100%, 0%);
    }
    &::after {
      width: 40px;
      height: 30px;
      left: 100%;
      top: 50%;
      transform: translate(-70%, 0%);
    }
    & ${Cloudpart}:nth-of-type(1) {
      width: 60px;
      height: 45px;
      left: -100%;
      transform: translate(40%, 0%);
      &::after {
        width: 55px;
        height: 35px;
        left: 50%;
        transform: translate(-35%, -50%);
      }
    }
    & ${Cloudpart}:nth-of-type(2) {
      width: 40px;
      height: 40px;
      left: 100%;
      transform: translate(-50%, 0%);
      &::before {
        width: 45px;
        height: 40px;
        left: -50%;
        transform: translate(20%, -70%);
      }
    }
  }
  &:nth-of-type(2) {
    width: 55px;
    height: 50px;
    left: -90px;
    top: 26px;
    animation: ${css `
      ${cloudtravel2} 18s linear infinite
    `};
    animation-delay: 1.2s;
    &::before {
      width: 30px;
      height: 20px;
      left: -55%;
      top: 70%;
      transform: translate(100%, -30%);
    }
    &::after {
      width: 40px;
      height: 30px;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -40%);
    }
    & ${Cloudpart}:nth-of-type(1) {
      width: 55px;
      height: 45px;
      left: -100%;
      transform: translate(50%, 0%);
      &::after {
        width: 55px;
        height: 30px;
        left: 50%;
        transform: translate(-35%, -20%);
      }
    }
    & ${Cloudpart}:nth-of-type(2) {
      width: 50px;
      height: 40px;
      left: 100%;
      transform: translate(-50%, 0%);
      &::before {
        width: 50px;
        height: 35px;
        left: -50%;
        transform: translate(10%, -40%);
      }
    }
  }
  &:nth-of-type(3) {
    width: 50px;
    height: 40px;
    left: -100px;
    top: 48px;
    animation: ${css `
      ${cloudtravel3} 30s linear infinite
    `};
    animation-delay: 8.6s;
    &::before {
      width: 40px;
      height: 30px;
      left: -50%;
      top: 50%;
      transform: translate(25%, -30%);
    }
    &::after {
      width: 40px;
      height: 25px;
      left: 50%;
      top: 50%;
      transform: translate(5%, -60%);
    }
    & ${Cloudpart}::nth-of-type(1) {
      width: 45px;
      height: 35px;
      left: -50%;
      top: 50%;
      transform: translate(0%, -40%);
      &::before {
        width: 30px;
        height: 20px;
        left: -30%;
        top: 50%;
        transform: translate(0%, -60%);
      }
      &::after {
        width: 20px;
        height: 15px;
        left: 0%;
        top: 50%;
        transform: translate(0%, 10%);
      }
    }
    & ${Cloudpart}::nth-of-type(2) {
      width: 50px;
      height: 45px;
      left: 50%;
      top: 50%;
      transform: translate(10%, -20%);
      &::before {
        width: 35px;
        height: 25px;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
      }
      &::after {
        width: 35px;
        height: 25px;
        left: 60%;
        top: 50%;
        transform: translate(0%, -40%);
      }
    }
    & ${Cloudpart}::nth-of-type(3) {
      width: 55px;
      height: 40px;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -10%);
      &::before {
        width: 40px;
        height: 25px;
        left: -50%;
        top: 50%;
        transform: translate(40%, -30%);
      }
    }
  }
`;
