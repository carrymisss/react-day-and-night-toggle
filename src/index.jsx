import React, { memo, useEffect, useState } from 'react';
import { Circle, Cloud, Cloudpart, Clouds, ContentWrapper, ShootingStar, ShootingStarWrapper, Star, Stars, Switch, SwitchContainer, } from './styles';
export const DayAndNightToggle = memo(({ onChange, checked, size = 32, startInactive = false, animationInactive = true, shadows = true, className = undefined, ...labelProps }) => {
    const [disable, setDisable] = useState(false);
    const [startingInactive, setStartingInactive] = useState(startInactive);
    const roundedSize = Math.floor(size);
    useEffect(() => {
        if (startingInactive) {
            setDisable(true);
            setTimeout(() => {
                setDisable(false);
            }, 2000);
        }
        return setStartingInactive(false);
    }, [checked, startingInactive]);
    const handleClick = () => {
        if (!disable) {
            onChange();
            setDisable(true);
            if (animationInactive) {
                setTimeout(() => {
                    setDisable(false);
                }, 2000);
            }
            else {
                setDisable(false);
            }
        }
    };
    return (<SwitchContainer className={className} size={roundedSize} {...labelProps}>
        <Switch shadows={shadows} size={roundedSize} checked={checked} onClick={handleClick}>
          <ContentWrapper>
            <Circle />
            <Stars>
              <Star />
              <Star />
              <Star />
              <ShootingStarWrapper>
                <ShootingStar />
              </ShootingStarWrapper>
            </Stars>
            <Clouds>
              <Cloud>
                <Cloudpart />
                <Cloudpart />
              </Cloud>
              <Cloud>
                <Cloudpart />
                <Cloudpart />
              </Cloud>
              <Cloud>
                <Cloudpart />
                <Cloudpart />
                <Cloudpart />
              </Cloud>
            </Clouds>
          </ContentWrapper>
        </Switch>
      </SwitchContainer>);
});
// DayAndNightToggle.propTypes = {
//     onChange: PropTypes.func.isRequired,
//     checked: PropTypes.bool.isRequired,
//     size: PropTypes.number,
//     startInactive: PropTypes.bool,
//     animationInactive: PropTypes.bool,
//     shadows: PropTypes.bool,
//     className: PropTypes.string,
// }
// DayAndNightToggle.defaultProps = {
//     onChange: () => null,
//     checked: false,
//     size: 32,
//     startInactive: false,
//     animationInactive: true,
//     shadows: true,
//     className: undefined,
// }
