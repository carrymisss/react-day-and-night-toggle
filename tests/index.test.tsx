import React from 'react';
import renderer from 'react-test-renderer';
import { beforeEach, describe, expect, test, vi } from 'vitest';

import { DayAndNightToggle } from '../src/index';

describe('Test of component <DayAndNightToggle>:', () => {
  let theme: boolean;
  let changeTheme: ReturnType<typeof vi.fn>;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    theme = false;
    changeTheme = vi.fn(() => (theme = !theme));
    renderer.act(() => {
      component = renderer.create(<DayAndNightToggle checked={theme} onChange={changeTheme} />);
    });
  });

  test('should render:', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('should have all the necessary props:', () => {
    expect(component.root.props).toEqual({ checked: theme, onChange: changeTheme });
    expect(changeTheme).toHaveBeenCalled;
  });
  test('should all props be passed:', () => {
    renderer.act(() => {
      component.update(
        <DayAndNightToggle
          checked={theme}
          onChange={changeTheme}
          size={24}
          startInactive={true}
          animationInactive={true}
          shadows={true}
          className={'class_name'}
        />,
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
    expect(component.root.props).toEqual({
      checked: theme,
      onChange: changeTheme,
      size: 24,
      startInactive: true,
      animationInactive: true,
      shadows: true,
      className: 'class_name',
    });
    expect(changeTheme).toHaveBeenCalled;
  });
});
