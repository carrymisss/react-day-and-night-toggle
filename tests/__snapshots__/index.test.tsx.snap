// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component <DayAndNightToggle>: should render: 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": null,
              "props": {
                "className": "sc-fqkwJk ePdnmY",
              },
              "type": "div",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": null,
                  "props": {
                    "className": "sc-iGgVNO jhIIpU",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": null,
                  "props": {
                    "className": "sc-iGgVNO jhIIpU",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": null,
                  "props": {
                    "className": "sc-iGgVNO jhIIpU",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-kAycRU etyDCI",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-gsFSjX kfhihz",
                  },
                  "type": "div",
                },
              ],
              "props": {
                "className": "sc-dcJtft",
              },
              "type": "div",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-dhKdPU dxxEnp",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-dhKdPU dxxEnp",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-dhKdPU dxxEnp",
                  },
                  "type": "div",
                },
              ],
              "props": {
                "className": "sc-jXbVAB",
              },
              "type": "div",
            },
          ],
          "props": {
            "className": "sc-eqUzNf beLJUp",
          },
          "type": "div",
        },
      ],
      "props": {
        "checked": false,
        "className": "sc-gEvDqW dhGSst",
        "onClick": [Function],
        "shadows": true,
        "size": 32,
      },
      "type": "div",
    },
  ],
  "props": {
    "className": "sc-aYaIB dZxTuZ",
    "size": 32,
  },
  "type": "div",
}
`;

exports[`Test of component <DayAndNightToggle>: should all props be passed: 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": null,
              "props": {
                "className": "sc-fqkwJk ePdnmY",
              },
              "type": "div",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": null,
                  "props": {
                    "className": "sc-iGgVNO jhIIpU",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": null,
                  "props": {
                    "className": "sc-iGgVNO jhIIpU",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": null,
                  "props": {
                    "className": "sc-iGgVNO jhIIpU",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-kAycRU etyDCI",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-gsFSjX kfhihz",
                  },
                  "type": "div",
                },
              ],
              "props": {
                "className": "sc-dcJtft",
              },
              "type": "div",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-dhKdPU dxxEnp",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-dhKdPU dxxEnp",
                  },
                  "type": "div",
                },
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                    {
                      "$$typeof": Symbol(react.test.json),
                      "children": null,
                      "props": {
                        "className": "sc-imWZod",
                      },
                      "type": "div",
                    },
                  ],
                  "props": {
                    "className": "sc-dhKdPU dxxEnp",
                  },
                  "type": "div",
                },
              ],
              "props": {
                "className": "sc-jXbVAB",
              },
              "type": "div",
            },
          ],
          "props": {
            "className": "sc-eqUzNf beLJUp",
          },
          "type": "div",
        },
      ],
      "props": {
        "checked": false,
        "className": "sc-gEvDqW glhzUP",
        "onClick": [Function],
        "shadows": true,
        "size": 24,
      },
      "type": "div",
    },
  ],
  "props": {
    "className": "sc-aYaIB hnbNJo class_name",
    "size": 24,
  },
  "type": "div",
}
`;
